File: Python/001~050/003. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if not s: return 0
        left, res, m = -1, 0, [-1 for i in range(256)]
        for i, v in enumerate(s):
            left = max(left, m[ord(v)])
            res = max(res, i-left)
            m[ord(v)] = i;
        return res
        

File: Go/003. Longest Substring Without Repeating Characters.go
func lengthOfLongestSubstring(str string) int {
	mp, res, left := make(map[rune]int), 0, -1
	for idx, c := range str {
        if _, ok := mp[c]; ok && mp[c] >= left {
			left = mp[c]
		}
        if idx-left > res {
            res = idx - left
        }
		mp[c] = idx
	}
	return res
}